doc/libpmem2/pmem2_config_new.3.md

```diff
@@ -0,0 +1,82 @@
+header: PMDK
+date: pmem2 API version 1.0
+...
+
+[comment]: <> (Copyright 2019, Intel Corporation)
+
+[comment]: <> (Redistribution and use in source and binary forms, with or without)
+[comment]: <> (modification, are permitted provided that the following conditions)
+[comment]: <> (are met:)
+[comment]: <> (    * Redistributions of source code must retain the above copyright)
+[comment]: <> (      notice, this list of conditions and the following disclaimer.)
+[comment]: <> (    * Redistributions in binary form must reproduce the above copyright)
+[comment]: <> (      notice, this list of conditions and the following disclaimer in)
+[comment]: <> (      the documentation and/or other materials provided with the)
+[comment]: <> (      distribution.)
+[comment]: <> (    * Neither the name of the copyright holder nor the names of its)
+[comment]: <> (      contributors may be used to endorse or promote products derived)
+[comment]: <> (      from this software without specific prior written permission.)
+
+[comment]: <> (THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS)
+[comment]: <> ("AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR)
+[comment]: <> (A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT)
+[comment]: <> (OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,)
+[comment]: <> (SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,)
+[comment]: <> (DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY)
+[comment]: <> (THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT)
+[comment]: <> ((INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE)
+[comment]: <> (OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.)
+
+[comment]: <> (pmem2_config_new.3 -- man page for pmem2_config_new and pmem2_config_delete
+
+[NAME](#name)<br />
+[SYNOPSIS](#synopsis)<br />
+[DESCRIPTION](#description)<br />
+[RETURN VALUE](#return-value)<br />
+[SEE ALSO](#see-also)<br />
+
+# NAME #
+
+**pmem2_config_new**(), **pmem2_config_delete**() - allocate and free a 
+configuration for a libpmem2 mapping
+
+# SYNOPSIS #
+
+```c
+#include <libpmem2.h>
+
+struct pmem2_config;
+int pmem2_config_new(struct pmem2_config **cfg);
+int pmem2_config_delete(struct pmem2_config **cfg);
+```
+
+# DESCRIPTION #
+
+The **pmem2_config_new**() function allocates *pmem2_config*, and places it in *\*cfg*. 
+Returned structure is initialized to default parameters.
+
+The **pmem2_config_delete**() function frees *\*cfg* returned by **pmem2_config_new**() and sets *\*cfg* to NULL. 
+
+# RETURN VALUE #
+
+The **pmem2_config_new**() function returns 0 on success or one of error values listed in the next section. 
+**pmem2_config_new**() does set *\*cfg* to NULL on failure.
```

Comment:
Why does it set *cfg* to NULL on failure? This is very uncommon.

Reply:
This is for convenience - you can either look at return value or *cfg to determine whether function failed or not.

Reply:
It's arguable that this is convenient.
IMHO we should always have one way of doing things, unless there are valid reasons otherwise (performance/efficiency like in the case of undo/redo transactions).